kernel/fs/iso9660.inc

```diff
@@ -736,3 +736,1044 @@ cd_compare_name:
         pop     esi eax edi edx
         stc
         ret
+
+;;; New driver
+
+DESCRIPTOR_TYPE_BOOT          = 0x00    ; not used
+DESCRIPTOR_TYPE_PRIMARY       = 0x01
+DESCRIPTOR_TYPE_SUPPLEMENTARY = 0x02
+; DESCRIPTOR_TYPE_PRIMARY for Juliet(UTF-16)
+DESCRIPTOR_TYPE_PARTITION     = 0x02
+DESCRIPTOR_TYPE_TERMANATOR    = 0xff    ; end scan descriptors
+
+struct  ISO9660_VOLUME_DESCRIPTOR
+        type    rb 1
+        magic   rb 5  ; 'CD001'
+        version rb 1
+ends
+
+struct  ISO9660_PRIMARY_DESCRIPTOR  ISO9660_VOLUME_DESCRIPTOR
+                        rb 1
+                        rb 32
+        VolumeName      rb 32 ; ascii or utf-16
+                        rb 8
+        VolumeSpaceSize rb 8
+                        rb 32
+                        rb 4 ; +120
+                        rb 4
+        LBA_Size        rb 4 ; +128
+        PathTableSize   rb 8         ; Do not use it!!!
+        LBA_PathTable   rb 4 ; +140  ; Do not use it!!!
+                        rb 4 ; +144  ; Do not use it!!!
```

Comment:
why? need clarification

Reply:
The file system assumes the use of 2 file search methods: using a table and using directory entry traversal. The driver uses the second option. These fields of the structure are necessary for the first search option.

Reply:
> The file system assumes the use of 2 file search methods: using a table and using directory entry traversal. The driver uses the second option. These fields of the structure are necessary for the first search option.

So you should say it in comment. Imagine you read the code of some big project and you run into comment like "Don't do it!!!" (without explanation) or like that. Is it okay?